operator/validator/controller.go

```diff
@@ -456,13 +459,18 @@ func (c *controller) StartValidators() {
 		return
 	}
 
+	mySubnets := c.selfSubnets()
+
 	var ownShares []*ssvtypes.SSVShare
-	var allPubKeys = make([][]byte, 0, len(shares))
+	var pubKeysToFetch [][]byte
 	for _, share := range shares {
 		if c.operatorDataStore.GetOperatorID() != 0 && share.BelongsToOperator(c.operatorDataStore.GetOperatorID()) {
 			ownShares = append(ownShares, share)
 		}
```

Comment:
wouldn't this do the same as the lines below? just use all the pubkeys from ownshares?

Reply:
@y0sher no, we need `ownShares` to start only own validators and we need `pubKeysToFetch` to fetch metadata

Reply:
yeah I was just trying to say, aren't we collecting it twice? aren't all the pubkeys to fetch already inside ownshares?

Reply:
I see it now. ownShares is not enough because there might be other committees in our subnets even though we are not part of them. resolving this.